pkgs/tools/X11/sunpaper/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, sunwait
+, wallutils
+, rPackages
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sunpaper";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    name = "sunpaper";
+    owner = "hexive";
+    repo = "sunpaper";
+    rev = "8d518dfddb5e80215ef3b884ff009df1d4bb74c2";
+    sha256 = "sCG7igD2ZwfHoRpR3Kw7dAded4hG2RbMLR/9nH+nZh8=";
+  };
+
+  buildInputs = with pkgs; [
+    wallutils
+    sunwait
+  ];
+
+  postPatch = ''
+    substituteInPlace sunpaper.sh \
+      --replace '$HOME/sunpaper/images/' "$out/images/"
+    substituteInPlace sunpaper.sh \
+      --replace '$HOME/.cache' "$TMPDIR/.sunpapercache"
+    substituteInPlace sunpaper.sh \
+      --replace "sunwait" "${sunwait}/bin/sunwait"
+    substituteInPlace sunpaper.sh \
+      --replace "setwallpaper" "${wallutils}/bin/setwallpaper"
+    '';
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mkdir -p "$out/images"
+    mkdir -p "$TMPDIR/.sunpapercache"
+    cp sunpaper.sh $out/bin/sunpaper
+    cp -R images $out
+  '';
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    $out/bin/sunpaper --help > /dev/null
+  '';
+
+  meta = with lib; {
+    description = "A utility to change wallpaper based on local weather, sunrise and sunset times.";
+    homepage = "https://github.com/hexive/sunpaper";
+    license = licenses.free;
```

Comment:
I'm afraid it'd be `unfree` by default if you can't get upstream to add a license.